PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-java-sdk/src/main/java/com/microsoft/azure/toolkit/intellij/java/sdk/utils/RuleConfigLoader.java

```diff
@@ -0,0 +1,247 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.azure.toolkit.intellij.java.sdk.utils;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.microsoft.azure.toolkit.intellij.java.sdk.models.RuleConfig;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class RuleConfigLoader {
+    private static final String CONFIG_FILE_PATH = "./META-INF/ruleConfigs.json";
+    private static final Logger LOGGER = Logger.getLogger(RuleConfigLoader.class.getName());
+    private static final RuleConfigLoader INSTANCE;
+
+    private final Map<String, RuleConfig> ruleConfigs;
+
+    static {
+        RuleConfigLoader tempInstance;
+        try {
+            tempInstance = new RuleConfigLoader(CONFIG_FILE_PATH);
```

Comment:
Generally, these things should be initialized when the IDE starts and not when the rule config is first used. This will block the editor window until the config is loaded.

Reply:
Attempted to add this to <poststartupactivity>, but it caused testing issues due to static initialization blocks. Could this be a blocker?

Reply:
Moved to startup activity tpo load the configuration once https://github.com/microsoft/azure-tools-for-java/pull/9654/commits/e275a81714af1ec65f2425dab76c027ee16150b0